batteryInterface.h

```diff
@@ -43,15 +39,25 @@ class TempratureIndicator
 	static float temperatureMinThreshould;
 	static float temperatureMaxThreshould;
 
-	bool temperatureStatus();
+	void temperatureStatus();
 	
 };
-class BatteryIndicator :public BatteryElements
+class BatteryParameters :public BatteryElements
 {
 public:
-	BatteryIndicator() {}
-	BatteryIndicator(string temp, float vol, float curr);
-	
+	BatteryParameters() {}
+	void Batteryparameters(string temp, float vol, float curr);
+	virtual void BatteryParametersCheck(string parameterStatus1, string parameterStatus2, double actualValue, double minValue, double MaxVAlue) = 0;

```

Comment:
The checking and the fact that it's a range check are in the same function.
That would mean the `BatteryParameters` interface cannot be used for a different kind of check.

Reply:
could you please discuss this in the up coming recap.